src/qmeta/python/schema_migration.py

```diff
@@ -3,216 +3,109 @@
 
         Returns
         -------
-        Integer number
+        version : `int` or ``Uninitialized``
+            The current schema version.
         """
 
+        # If the `qservMeta` database does not exist then QMeta is
+        # Uninitialized.
+        cursor = self.connection.cursor()
+        cursor.execute("SHOW DATABASES")
+        if ('qservMeta',) not in cursor.fetchall():
+            return Uninitialized
+
         # Initial QMeta implementation did not have version number stored at all,
         # and we call this version 0. Since version=1 version number is stored in
         # QMetadata table with key="version"
-        if not utils.tableExists(self.engine, "QMetadata"):
-            _log.debug("QMetadata missing: version=0")
+        cursor.execute("select 1 from INFORMATION_SCHEMA.TABLES where TABLE_NAME = 'QMetadata';")
+        if not cursor.fetchone():
             return 0
-        else:
-            query = "SELECT value FROM QMetadata WHERE metakey = 'version'"
-            result = self.engine.execute(query)
-            row = result.first()
-            if row:
-                _log.debug("found version in database: %s", row[0])
-                return int(row[0])
-
-    def latest_version(self):
-        """Returns latest known schema version.
 
-        Returns
-        -------
-        Integer number
-        """
-        if self.scripts:
-            version = max(script[1] for script in self.scripts)
-            _log.debug("latest version from migration scripts: %s", version)
-        else:
-            # no migration scripts - current version is latest
-            version = self.current_version()
-            _log.debug("no migration scripts, returning current version %s", version)
-        return version
-
-    def migrations(self):
-        """Returns all known migrations.
-
-        Returns
-        -------
-        List of tuples, each tuple contains three elements:
-        - starting schema version (int)
-        - final schema version (usually starting + 1)
-        - migration script name (or any arbitrary string)
-        """
-        return [script[:3] for script in self.scripts]
+        self.connection.database = qservMetaDb
+        cursor = self.connection.cursor()
+        cursor.execute("SELECT value FROM QMetadata WHERE metakey = 'version'")
+        result = cursor.fetchone()
+        if not result:
+            return Uninitialized
+        return int(result[0])
+
+    @backoff.on_exception(
+        exception=(mysql.connector.errors.OperationalError, mysql.connector.errors.ProgrammingError),
+        wait_gen=backoff.expo,
+        on_backoff=on_backoff(log=_log),
+        max_time=max_backoff_sec,
+    )
+    def _set_version(self, version):
+        """Set the version number stored in QMetadata."""
+        # make sure that current version is updated in database
+        query = "UPDATE QMetadata SET value = {} WHERE metakey = 'version'".format(version)
```

Comment:
I believe `version` is an `int` here but `QMetadata.value` is a string. Mysql is very liberal when mixing types, but it may be better to convert version to string before saving, e.g.:

             query = f"UPDATE QMetadata SET value = '{version}' WHERE metakey = 'version'"